model-optimizer/extensions/middle/EltwiseInputReshape.py

```diff
@@ -65,58 +67,41 @@ def find_and_replace_pattern(self, graph: Graph):
 
 
 class EltwiseInputReshape(MiddleReplacementPattern):
-    enabled = True
-    force_clean_up = True
-
-    def run_after(self):
-        from extensions.middle.pass_separator import MiddleStart
-        return [MiddleStart]
+    # This pass should be called directly from pipeline before layout change and other permutations
+    enabled = False
+    force_shape_inference = True
```

Comment:
`force_shape_inference` does not make sense because the transformation is disabled.

Reply:
Looks like we need to restore the original solution with calling `infer` method for the ops which will change the output shape.

Reply:
Fixed.